mono/utils/memfuncs.c

```diff
@@ -273,6 +273,10 @@ mono_determine_physical_ram_size (void)
 #elif defined (HAVE_SYSCONF)
 	guint64 page_size = 0, num_pages = 0;
 
+	gint64 restricted_limit = getRestrictedPhysicalMemoryLimit();
```

Comment:
coreclr gc sets the managed heap size to a 75% of the container memory size. The other 25% is reserved for 'native' memory.

The limit is controllable through envvars: https://docs.microsoft.com/en-us/dotnet/core/run-time-config/garbage-collector#heap-limit.